__tests__/index.spec.js

```diff
@@ -392,6 +392,17 @@ pluginTester({
       title: "Should merge single className and a style with a function call",
       code:
         "const Foo = () => <div className={styles.foo} style={myFn()}>Foo</div>"
+    },
+    {
+      title:
+        "Should support multiple classnames type in array styles objects",
+      code: `
+        const 
+          rd = Math.random(10),
+          style1 = _reactNativeDynamicStyleProcessor.process(styles).bar,
```

Comment:
Sorry, I don't think that explained very well what the needed change was.

The `_reactNativeDynamicStyleProcessor.process(...)` needs to be part of the generated Babel output, not the unit test. So currently the generated code looks correct, but the unit test should be what it was previously.

That means that some changes are still required in `index.js` to make the code behave correctly and the Babel output to look like it does right now.

Reply:
Sorry, I don't know how to do that.

Reply:
Don't worry, I can have a look at that and see if there is any easy way to fix it.

Reply:
```
        import styles from "./styles.css";
        const 
          rd = Math.random(10),
          style1 = styles.bar,
          style2 = rd % 2 === 1 ? styles.bay : styles.baz,
          Foo = () => <div className={[styles.foo, style1, style2].join(' ')}>Bar</div>
```
It's become wrong to : 

```var _styles = require(\\"./styles.css\\");

var _styles2 = _interopRequireDefault(_styles);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _reactNativeDynamicStyleProcessor = require(\\"react-native-dynamic-style-processor\\");

var rd = Math.random(10),
    style1 = _styles2.default.bar,
    style2 = rd % 2 === 1 ? _styles2.default.bay : _styles2.default.baz,
    Foo = function Foo() {
  return React.createElement(
    \\"div\\",
    { style: [_reactNativeDynamicStyleProcessor.process(_styles2.default).foo, style1, style2] },
    \\"Bar\\"
  );
};```

Out of the `className` attribute, `styles` variable does'nt transform to `_reactNativeDynamicStyleProcessor(styles)`.

Reply:
I can do that, but I'm not sure that what you want : 

```
        const
          _reactNativeDynamicStyleProcessor = require('react-native-dynamic-style-processor'),
          rd = Math.random(10),
          style1 = _reactNativeDynamicStyleProcessor.process(styles).bar,
          style2 = rd % 2 === 1 ? _reactNativeDynamicStyleProcessor.process(styles).bay : _reactNativeDynamicStyleProcessor.process(styles).baz,
          Foo = () => <div className={[_reactNativeDynamicStyleProcessor.process(styles).foo, style1, style2].join(' ')}>Bar</div>
```
===>>>
```var _reactNativeDynamicStyleProcessor = require('react-native-dynamic-style-processor'),
    rd = Math.random(10),
    style1 = _reactNativeDynamicStyleProcessor.process(styles).bar,
    style2 = rd % 2 === 1 ? _reactNativeDynamicStyleProcessor.process(styles).bay : _reactNativeDynamicStyleProcessor.process(styles).baz,
    Foo = function Foo() {
  return React.createElement(
    'div',
    { style: [_reactNativeDynamicStyleProcessor2.process(_reactNativeDynamicStyleProcessor.process(styles)).foo, style1, style2] },
    'Bar'
  );
};

var _reactNativeDynamicStyleProcessor2 = require('react-native-dynamic-style-processor');```

Reply:
This is the correct unit test:

```jsx
        import styles from "./styles.css";
        const 
          rd = Math.random(10),
          style1 = styles.bar,
          style2 = rd % 2 === 1 ? styles.bay : styles.baz,
          Foo = () => <div className={[styles.foo, style1, style2].join(' ')}>Bar</div>
```

...and this is what the output from Babel should look like:

```js
var _reactNativeDynamicStyleProcessor = require('react-native-dynamic-style-processor');
var rd = Math.random(10),
    style1 = _reactNativeDynamicStyleProcessor.process(styles).bar,
    style2 = rd % 2 === 1 ? _reactNativeDynamicStyleProcessor.process(styles).bay : _reactNativeDynamicStyleProcessor.process(styles).baz,
    Foo = function Foo() {
  return React.createElement(
    'div',
    { style: [_reactNativeDynamicStyleProcessor.process(styles).foo, style1, style2] },
    'Bar'
  );
};
```

Reply:
https://github.com/kristerkari/babel-plugin-react-native-classname-to-dynamic-style/pull/7#discussion_r297150584

It's one of my test, but the output does'nt look like the same.

Reply:
> It's one of my test, but the output does'nt look like the same.

Babel adds a couple of more lines to the  `require` call to handle default exports, so my example is does not match the output 100%

Reply:
ok, so i will commit this test :)